keyboards/preonic/keymaps/alfrdmalr/keymap.c

```diff
@@ -0,0 +1,271 @@
+/* Copyright 2015-2017 Jack Humbert 2019-2020 Alfred Maler
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "muse.h"
+
+enum preonic_layers {
+  _QWERTY,
+  // _COLEMAK,
+  // _DVORAK,
+  _SETTINGS,
+  _SYMBOLS,
+  _NAVIGATION,
+  _NUMPAD,
+};
+
+enum preonic_keycodes {
+  QWERTY = SAFE_RANGE,
+  // COLEMAK,
+  // DVORAK,
+  // BACKLIT,
+  SYMBOLS,
+  NAVIGATION,
+  NUMPAD,
+  SETTINGS
+};
+
+#define NUMSPACE LT(_NUMPAD, KC_SPC)
+#define CTRLSHFT C(KC_LSFT)
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+/* Qwerty
+ * ,-----------------------------------------------------------------------------------.
+ * |   `  |   1  |   2  |   3  |   4  |   5  |   6  |   7  |   8  |   9  |   0  | Bksp |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | Tab  |   Q  |   W  |   E  |   R  |   T  |   Y  |   U  |   I  |   O  |   P  | Del  |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | NAV  |   A  |   S  |   D  |   F  |   G  |   H  |   J  |   K  |   L  |   ;  |  "   |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | SHFT |   Z  |   X  |   C  |   V  |   B  |   N  |   M  |   ,  |   .  |   /  | SHFT |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | CTRL | C/S  | LGUI | LALT | SYMB |  SPACE/NUM  | SYMB | RALT | SETT |mu tog| LEAD |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_QWERTY] = LAYOUT_preonic_grid( \
+  KC_GRV,      KC_1,     KC_2,    KC_3,    KC_4,    KC_5,     KC_6,     KC_7,    KC_8,    KC_9,     KC_0,    KC_BSPC, \
+  KC_TAB,      KC_Q,     KC_W,    KC_E,    KC_R,    KC_T,     KC_Y,     KC_U,    KC_I,    KC_O,     KC_P,    KC_DEL,  \
+  NAVIGATION,  KC_A,     KC_S,    KC_D,    KC_F,    KC_G,     KC_H,     KC_J,    KC_K,    KC_L,     KC_SCLN, KC_QUOT, \
+  KC_LSFT,     KC_Z,     KC_X,    KC_C,    KC_V,    KC_B,     KC_N,     KC_M,    KC_COMM, KC_DOT,   KC_SLSH, KC_RSFT,  \
+  KC_LCTL,     CTRLSHFT, KC_LGUI, KC_LALT, SYMBOLS, NUMSPACE, NUMSPACE, SYMBOLS, KC_RALT, SETTINGS, MU_TOG,  KC_LEAD  \
+),
```

Comment:
The backslashes at the end of these lines are not necessary. You can keep them if you want as it isn't hurting anything, but IMO it looks neater without them.

Reply:
Thanks, looks much better without.